yjit/src/codegen.rs

```diff
@@ -4972,19 +4975,20 @@ fn gen_invokebuiltin(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock,
 
     // Call the builtin func (ec, recv, arg1, arg2, ...)
     mov(cb, C_ARG_REGS[0], REG_EC);
-    mov(cb, C_ARG_REGS[1], member_opnd(REG_CFP, rb_control_frame_t, self));
+    mov(cb, C_ARG_REGS[1], mem_opnd(64, REG_CFP, RUBY_OFFSET_CFP_SELF));
 
     // Copy arguments from locals
-    for (int32_t i = 0; i < bf->argc; i++) {
-        let stack_opnd = ctx.stack_opnd(bf->argc - i - 1);
-        let c_arg_reg = C_ARG_REGS[2 + i];
+    for i in 0..bf_argc {
+        let stack_opnd = ctx.stack_opnd(bf_argc - i - 1);
+        let idx: usize = (2 + i).try_into().unwrap();
+        let c_arg_reg = C_ARG_REGS[idx];
         mov(cb, c_arg_reg, stack_opnd);
     }
 
-    call_ptr(cb, REG0, (void *)bf->func_ptr);
+    call_ptr(cb, REG0, unsafe { (*bf).func_ptr } as *const u8);
 
     // Push the return value
-    ctx.stack_pop(bf->argc);
+    ctx.stack_pop(bf_argc.try_into().unwrap());
```

Comment:
We could use `as` here too because we've already checked that `bf_argc` is less than a small constant.